### PR TITLE
[fix](array-type) array can not be distributed key and aggregation key

### PR DESCRIPTION
**Pull request by cambyzju**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Array column should not be `distributed key` or `group by` or `aggregate key`, we should forbid it before create table.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/HashDistributionDesc.java [hunk 1]
```diff
@@ -80,6 +80,9 @@ public void analyze(Set<String> colSet, List<ColumnDef> columnDefs) throws Analy
                     if (columnDef.getType().isScalarType(PrimitiveType.STRING)) {
                         throw new AnalysisException("String Type should not be used in distribution column["
                                 + columnDef.getName() + "].");
+                    } else if (columnDef.getType().isArrayType()) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/HashDistributionDesc.java:83]: Maybe we should move all these checks to `toDistributionInfo()` ?

Comment (cambyzju) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/HashDistributionDesc.java]: done

**Review by xy720 — APPROVED** [review]
LGTM

**Review by xy720 — COMMENTED** [review]
We should also update the array docs to tell user don't use array type as distribution key.

**Comment by cambyzju** [comment]
> We should also update the array docs to tell user don't use array type as distribution key.
Already exists:
`An array of T-type items, it cannot be used as a key column.`

**Pull request merged**